src/crimson/os/seastore/cache.h

```diff
@@ -495,16 +497,22 @@ class Cache : public ExtentTransViewRetriever {
       } else {
         // stable from trans-view
         assert(!p_extent->is_pending_in_trans(t.get_trans_id()));
-        if (t.maybe_add_to_read_set(p_extent)) {
+        auto ret = t.maybe_add_to_read_set(p_extent);
+        if (ret.added) {
           if (p_extent->is_dirty()) {
             ++access_stats.cache_dirty;
             ++stats.access.cache_dirty;
           } else {
             ++access_stats.cache_lru;
             ++stats.access.cache_lru;
           }
-          touch_extent(*p_extent, &t_src, t.get_cache_hint());
+          if (ret.is_paddr_known) {
```

Comment:
If reasonable, add comments, e.g.:
```
// FIXME: retired-placeholder isn't linked in the lba tree yet.
//   We think it's still working because:
//   1. A retired-placeholder must be an ObjectDataBlock
//   2. Per rados object, no read is possible during write, and write cannot be parallel
assert(!is_retired_placeholder_type(p_extent->get_type()));